Region/Histogram.cc

```diff
@@ -36,15 +36,15 @@ void Histogram::join(Histogram& h) { // NOLINT
     tbb::parallel_for(range, loop);
 }
 
-void Histogram::setup_bins(const size_t start, const size_t end) {
-    size_t i, stride, buckets;
+void Histogram::setup_bins() {
+    int64_t i, stride, buckets;
     int** bins_bin;
 
     auto calc_lambda = [&](size_t start, size_t lstride) {
         int* lbins = new int[_hist.size()];
         size_t end = std::min((size_t)(start + lstride), _data.size());
         memset(lbins, 0, _hist.size() * sizeof(int));
-        for (size_t i = start; i < end; i++) {
+        for (auto i = start; i < end; i++) {
```

Comment:
since `start` and `end` are both `size_t`, won't the auto end up as `size_t` as well? Although I suppose since it's not in a parallel for, it's fine